corehq/apps/userreports/static/userreports/js/builder_view_models.js

```diff
@@ -185,56 +200,157 @@ hqDefine('userreports/js/builder_view_models.js', function () {
         return columnsValid && columnLengthValid;
     };
     PropertyList.prototype.buttonHandler = function () {
-        this.columns.push(new PropertyListItem());
+        this.columns.push(new PropertyListItem(
+            this.getDefaultDisplayText.bind(this),
+            this.getPropertyObject.bind(this)
+        ));
         if (!_.isEmpty(this.analyticsAction) && !_.isEmpty(this.analyticsLabel)){
             window.analytics.usage("Report Builder", this.analyticsAction, this.analyticsLabel);
             window.analytics.workflow("Clicked " + this.analyticsAction + " in Report Builder");
         }
     };
+    /**
+     * Return the default display text for this property. For questions, it is
+     * the question label.
+     * @param {string} property_id
+     * @returns {string}
+     */
+    PropertyList.prototype.getDefaultDisplayText = function (property_id) {
+        var property = this.getPropertyObject(property_id);
+        if (property !== undefined) {
+            // property.display will exist if the property is a ColumnOption
+            // property.text will exist if the property is a DataSourceProperty
+            return property.display || property.text || property_id;
+        }
+        return property_id;
+    };
+    /**
+     * Return the object representing the property corresponding to the given
+     * property_id.
+     * @param {string} property_id
+     * @returns {object}
+     */
+    PropertyList.prototype.getPropertyObject = function(property_id) {
+        // TODO: This is wrong as shit because we need to match on id sometimes.
```

Comment:
:hankey: Assuming this will be fixed in a future PR? looks the code was already here anyways